### PR TITLE
HPCC-26380 Roxie topology server heartbeat times are too short 

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ]  My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: helm/hpcc/templates/roxie.yaml [hunk 1]
```diff
@@ -62,7 +63,7 @@ data:
     version: 1.0
     toposerver:
       port: {{ .topoport }}
-      traceLevel: {{ .toposerver.traceLevel | default 1 }}
+{{ toYaml ( omit .toposerver "port" "logging" "env") | indent 6 }}
```
Comment (ghalliday) [inline helm/hpcc/templates/roxie.yaml:66]: Should this be "topoport" instead of "port"?

File: roxie/topo/toposerver.cpp [hunk 2]
```diff
@@ -357,7 +361,15 @@ int main(int argc, const char *argv[])
...
-#ifndef _CONTAINERIZED
+
+        timeoutCheckInterval = topology->getPropInt("@timeoutCheckInterval", timeoutCheckInterval);
+        heartbeatInterval = topology->getPropInt("@heartbeatInterval", heartbeatInterval);
+        timeoutHeartbeatAgent = topology->getPropInt("@timeoutHeartbeatServer", timeoutHeartbeatAgent);
+        timeoutHeartbeatServer = topology->getPropInt("@timeoutHeartbeatServer", timeoutHeartbeatServer);
+        removeHeartbeatInterval = topology->getPropInt("@removeHeartbeatInterval", removeHeartbeatInterval);
+        topologyReportInterval = topology->getPropInt("@topologyReportInterval", topologyReportInterval);
+
+        #ifndef _CONTAINERIZED
```
Comment (ghalliday) [inline roxie/topo/toposerver.cpp:372]: trivial: indentation

File: roxie/topo/toposerver.cpp [hunk 3]
```diff
@@ -357,7 +361,15 @@ int main(int argc, const char *argv[])
         Owned<IPropertyTree> topology = loadConfiguration(defaultYaml, argv, "toposerver", "TOPOSERVER", nullptr, nullptr);
         traceLevel = topology->getPropInt("@traceLevel", 1);
         topoPort = topology->getPropInt("@port", TOPO_SERVER_PORT);
-#ifndef _CONTAINERIZED
+
+        timeoutCheckInterval = topology->getPropInt("@timeoutCheckInterval", timeoutCheckInterval);
+        heartbeatInterval = topology->getPropInt("@heartbeatInterval", heartbeatInterval);
+        timeoutHeartbeatAgent = topology->getPropInt("@timeoutHeartbeatServer", timeoutHeartbeatAgent);
```
Comment (ghalliday) [inline roxie/topo/toposerver.cpp:367]: Typo: timeoutHeartbeatServer =-> timeoutHeartbeatAgent

File: roxie/udplib/udptopo.cpp [hunk 4]
```diff
@@ -483,12 +501,13 @@ extern UDPLIB_API void createStaticTopology(const std::vector<RoxieEndpointInfo>
 
 static std::thread topoThread;
 static Semaphore abortTopo;
-const unsigned topoUpdateInterval = 5000;
+unsigned heartbeatInterval = 10000;   // How often roxie servers update topo server
```
Comment (ghalliday) [inline roxie/udplib/udptopo.cpp:504]: Other places have 5000 as the default - also line 510.

**Review by ghalliday — CHANGES_REQUESTED** [review]
@richardkchapman a few comments.  Should also squash the two commits.

File: helm/hpcc/templates/roxie.yaml (hunk 1 above)
Comment (richardkchapman) [reply helm/hpcc/templates/roxie.yaml:66]: I don't think so. .topoport is the value of .toposerver.port copied into roxie level

I could remove the "port" and the line above - would probably be cleaner

**Review by ghalliday — APPROVED** [review]

**Pull request merged**